emsdk.py

```diff
@@ -322,7 +322,10 @@ def win_set_environment_variable_direct(key, value, system=True):
       folder.Close()
 
 
-def win_get_environment_variable(key, system=True, fallback=True):
+def win_get_environment_variable(key, system=True, user=True, fallback=True):
+  if (not system and not user and fallback):
+    # if no --system or --permanent flag is provided use shell's value
+    return os.environ[key]
```

Comment:
Instead of this change, I think we need call `win_get_environment_variable` only when constructing the path to store in the registry.

When constructing the path for the current environ we can simply not call `win_get_environment_variable` at all.  See my comment below.

Reply:
`win_get_environment_variable` is a function for getting environment variables. It has options for making it shell only. When we do not pass `--permanent` or `--system` it becomes shell only. That's how we determine shell only situation.

I don't see any benefit in introducing a new function.